File: src/main/resources/rules/FractionsRule2.js
//Imports


var Error = org.warp.picalculator.Error;
var Function = org.warp.picalculator.math.Function;
var Division = org.warp.picalculator.math.functions.Division;
var Number = org.warp.picalculator.math.functions.Number;

var ObjectArrayList = Java.type("it.unimi.dsi.fastutil.objects.ObjectArrayList");

/**
 * Fractions rule
 * a / 1 = a
 * 
 * @author Andrea Cavalli
 *
 */
var rule = {
	// Rule name
	getRuleName: function() {
		return "FractionsRule2";
	},
	// Rule type
	getRuleType: function() {
		return RuleType.CALCULATION;
	},
	/* Rule function
	   Returns:
	     - null if it's not executable on the function "f"
	     - An ObjectArrayList<Function> if it did something
	*/

	execute: function(f) {
		var isExecutable;
		if (ScriptUtils.instanceOf(f, Division.class)) {
			var fnc = f;
			if (ScriptUtils.instanceOf(fnc.getParameter2(), Number.class)) {
				var numb = fnc.getParameter2();
				if (numb.equals(new Number(f.getMathContext(), 1))) {
					isExecutable = true;
				}
			}
		}
	
		if (isExecutable) {
			var result = new ObjectArrayList();
			var fnc = f;
			result.add(fnc.getParameter1());
			return result;
		} else {
			return null;
		}
	}
}

//Add this rule to the list of rules
RulesManager.addRule(engine.getInterface(rule, Rule.class));

File: src/main/resources/rules/FractionsRule2.java
/*
SETTINGS: (please don't move this part)
 PATH=FractionsRule2
*/

//Imports


import org.warp.picalculator.Error;
import org.warp.picalculator.math.Function;
import org.warp.picalculator.math.functions.Division;
import org.warp.picalculator.math.functions.Number;

import it.unimi.dsi.fastutil.objects.ObjectArrayList;

/**
 * Fractions rule
 * a / 1 = a
 * 
 * @author Andrea Cavalli
 *
 */
public class FractionsRule2 implements Rule {
	// Rule name
	@Override
	public String getRuleName() {
		return "FractionsRule2";
	}

	// Rule type
	@Override
	public RuleType getRuleType() {
		return RuleType.CALCULATION;
	}

	/* Rule function
	   Returns:
	     - null if it's not executable on the function "f"
	     - An ObjectArrayList<Function> if it did something
	*/

	@Override
	public ObjectArrayList<Function> execute(Function f) {
		boolean isExecutable = false;
		if (f instanceof Division) {
			Division fnc = (Division) f;
			if (fnc.getParameter2() instanceof Number) {
				Number numb = (Number) fnc.getParameter2();
				if (numb.equals(new Number(f.getMathContext(), 1))) {
					isExecutable = true;
				}
			}
		}
	
		if (isExecutable) {
			ObjectArrayList<Function> result = new ObjectArrayList<>();
			Division fnc = (Division) f;
			result.add(fnc.getParameter1());
			return result;
		} else {
			return null;
		}
	}
}
